doc/examples/motion_planning_python_api/scripts/motion_planning_python_api_tutorial.py

```diff
@@ -0,0 +1,181 @@
+    PlanRequestParameters,
+    MultiPipelinePlanRequestParameters,
+)
+from moveit.core.robot_state import RobotState
+from moveit.core.kinematic_constraints import construct_joint_constraint
+
+from ament_index_python.packages import get_package_share_directory
+from moveit_configs_utils import MoveItConfigsBuilder
+import yaml
+
+
+def main():
+
+    ###################################################################
+    # MoveItPy Setup
+    ###################################################################
+    rclpy.init()
+    logger = rclpy.logging.get_logger("moveit_py.pose_goal")
+
+    # instantiate MoveItPy instance and get planning component
+    moveit = MoveItPy(node_name="moveit_py")
+    panda_arm = moveit.get_planning_component("panda_arm")
+    logger.info("MoveItPy instance created")
+
+    ###########################################################################
+    # Plan 1 - set states with predefined string
+    ###########################################################################
+
+    # set plan start state using predefined state
+    panda_arm.set_start_state(configuration_name="ready")
+
+    # set pose goal using predefined state
+    panda_arm.set_goal_state(configuration_name="extended")
+
+    # plan to goal
+    logger.info("Planning trajectory")
+    plan_result = panda_arm.plan()
+
+    # execute the plan
+    if plan_result:
+        logger.info("Executing plan")
+        robot_trajectory = plan_result.trajectory
+        moveit.execute(robot_trajectory, blocking=True, controllers=[])
+
+    ###########################################################################
+    # Plan 2 - set goal state with RobotState object
+    ###########################################################################
+
+    # instantiate a RobotState instance using the current robot model
+    robot_model = moveit.get_robot_model()
+    robot_state = RobotState(robot_model)
+
+    # randomize the robot state
+    robot_state.set_to_random_positions()
+
+    # set plan start state to current state
+    panda_arm.set_start_state_to_current_state()
+
+    # set goal state to the initialized robot state
+    logger.info("Set goal state to the initialized robot state")
+    panda_arm.set_goal_state(robot_state=robot_state)
+
+    # plan to goal
+    logger.info("Planning trajectory")
+    plan_result = panda_arm.plan()
+
+    # execute the plan
+    if plan_result:
+        logger.info("Executing plan")
+        robot_trajectory = plan_result.trajectory
+        moveit.execute(robot_trajectory, blocking=True, controllers=[])
+
+    ###########################################################################
+    # Plan 3 - set goal state with PoseStamped message
+    ###########################################################################
+
+    # set plan start state to current state
+    panda_arm.set_start_state_to_current_state()
+
+    # set pose goal with PoseStamped message
+    pose_goal = PoseStamped()
```

Comment:
I might suggest importing the `PoseStamped` message in this block itself so it's clearer to users where this comes from, rather than at the top.

If the linters don't like this, then ignore this comment.

Reply:
Also, same with other things as needed, like the joint constraints, etc.

Reply:
Ok I will make this change.